### PR TITLE
En 1230 - add save ad delete widget api 

### PR DESCRIPTION
**Pull request by fedelocci**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: test/api/widgets.test.js [hunk 1]
```diff
@@ -95,4 +97,63 @@ describe('api/widgets', () => {
...
+      jest.clearAllMocks();
+    });
+    it('returns a promise', () => {
+      expect(postWidgets()).toBeInstanceOf(Promise);
+    });
+
+    it('if successful, returns a mock ok response', () => {
+      postWidgets(WIDGET_POST_PUT);
+      expect(makeRequest).toHaveBeenCalledWith({
+        ...correctRequest,
```
Comment (alessioscalici) [inline test/api/widgets.test.js]: Why copying correctRequest, if you are overwriting all its properties?

File: test/api/widgets.test.js [hunk 2]
```diff
@@ -95,4 +97,63 @@ describe('api/widgets', () => {
...
+      jest.clearAllMocks();
+    });
+    it('returns a promise', () => {
+      expect(putWidgets()).toBeInstanceOf(Promise);
+    });
+
+    it('if successful, returns a mock ok response', () => {
+      putWidgets('AAA', WIDGET_POST_PUT);
+      expect(makeRequest).toHaveBeenCalledWith({
+        ...correctRequest,
```
Comment (alessioscalici) [inline test/api/widgets.test.js]: Why copying correctRequest, if you are overwriting all its properties?

File: test/api/widgets.test.js [hunk 3]
```diff
@@ -95,4 +97,63 @@ describe('api/widgets', () => {
...
+      jest.clearAllMocks();
+    });
+    it('returns a promise', () => {
+      expect(deleteWidgets()).toBeInstanceOf(Promise);
+    });
+
+    it('if successful, returns a mock ok response', () => {
+      deleteWidgets('AAA');
+      expect(makeRequest).toHaveBeenCalledWith({
+        ...correctRequest,
```
Comment (alessioscalici) [inline test/api/widgets.test.js]: Why copying correctRequest, if you are overwriting all its properties?

File: test/state/widgets/actions.test.js [hunk 4]
```diff
@@ -45,121 +47,217 @@ jest.mock('state/widgets/selectors', () => ({
...
+          expect(store.getActions()).toHaveLength(1);
+          done();
+        }).catch(done.fail);
+      });
+
+      it('if there is no widget selected, fetch and select the new widget', (done) => {
+        getSelectedWidget.mockReturnValue(null);
+        store.dispatch(loadSelectedWidget(WIDGET.code)).then(() => {
+          expect(getWidget).toHaveBeenCalled();
+          expect(store.getActions()).toHaveLength(1);
```
Comment (alessioscalici) [inline test/state/widgets/actions.test.js]: Why not checking which action has been dispatched?

File: test/state/widgets/actions.test.js [hunk 5]
```diff
@@ -45,121 +47,217 @@ jest.mock('state/widgets/selectors', () => ({
...
+          expect(getWidget).not.toHaveBeenCalled();
+          expect(store.getActions()).toHaveLength(0);
+          done();
+        }).catch(done.fail);
+      });
+
+      it('if there is another widget selected, fetch and select the new widget', (done) => {
+        store.dispatch(loadSelectedWidget('some_other_widget')).then(() => {
+          expect(getWidget).toHaveBeenCalled();
+          expect(store.getActions()).toHaveLength(1);
```
Comment (alessioscalici) [inline test/state/widgets/actions.test.js]: Why not checking which action has been dispatched?

File: test/state/widgets/actions.test.js [hunk 6]
```diff
@@ -45,121 +47,217 @@ jest.mock('state/widgets/selectors', () => ({
...
+      it('calls setWidgetList and setPage action', (done) => {
+        getWidgets.mockImplementation(mockApi({ payload: WIDGET_LIST }));
+        store.dispatch(fetchWidgetList()).then(() => {
+          expect(getWidgets).toHaveBeenCalled();
+          const actions = store.getActions();
+          expect(actions).toHaveLength(4);
+          expect(actions[0]).toHaveProperty('type', TOGGLE_LOADING);
+          expect(actions[1]).toHaveProperty('type', SET_WIDGET_LIST);
+          expect(actions[1]).toHaveProperty('payload');
+          expect(actions[2].type).toEqual(SET_PAGE);
```
Comment (alessioscalici) [inline test/state/widgets/actions.test.js]: Why not using toHaveProperty here, like you did in other assertions?

File: test/test/mocks/widgets.js [hunk 7]
```diff
@@ -98,21 +51,22 @@ export const WIDGET_LIST = {
...
-      titles: {
-        it: 'Mio Widget',
-        en: 'My Widget',
-      },
-    },
-  ],
+  'User Widget': [WIDGET],
+};
+
+export const WIDGET_POST_PUT = {
```
Comment (alessioscalici) [inline test/test/mocks/widgets.js]: What is the meaning of this POST_PUT suffix? Can't this mock be replaced by WIDGET in the files that are using it?

**Comment by entando-service** [comment]
All is well

**Pull request merged**